src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/azclishell/az_completer.py

```diff
@@ -136,8 +140,13 @@ def get_completions(self, document, complete_event):
         if not self.started:
             return
 
-        text = document.text_before_cursor
-        text = self.reformat_cmd(text)
+        text = self.reformat_cmd(document.text_before_cursor)
+        event_payload = {
+            'text': text
+        }
+        self.shell_ctx.cli_ctx.raise_event(EVENT_INTERACTIVE_PRE_SUB_TREE_CREATE, event_payload=event_payload)
```

Comment:
Consider changing the name of this now that its location has changed.
Maybe something like `EVENT_INTERACTIVE_PRE_COMPLETER_TEXT_PARSING`